promitor-agent-resource-discovery/templates/deployment.yaml

```diff
@@ -56,9 +56,9 @@ spec:
             - name: http
               containerPort: {{ .Values.service.targetPort }}
               protocol: TCP
-        {{- if or (or .Values.azureAuthentication.identity.key .Values.azureAuthentication.appKey) .Values.deployment.env.extra }}
+  {{- if or (or .Values.azureAuthentication.identity.key .Values.azureAuthentication.appKey (eq .Values.secrets.createSecret false)) .Values.deployment.env.extra }}
           env:
-        {{- if or .Values.azureAuthentication.identity.key .Values.azureAuthentication.appKey}}
+  {{- if or .Values.azureAuthentication.identity.key .Values.azureAuthentication.appKey (eq .Values.secrets.createSecret false) }}
```

Comment:
Don't we need to imrpove this because `or {criteria} {criteria} ({criteria})`

Reply:
I think it is correct, and it works as intended against my fork. It will be true if either identity.key or appKey is not nil/empty value or if createSecret equals false.

Reply:
Sounds good to me, thanks!